kartothek/io_components/metapartition.py

```diff
@@ -1348,6 +1349,14 @@ def _partition_data(self, partition_on):
         ]
         dct = dict()
         empty_tables = []
+
+        performing_safety_check = False
```

Comment:
Is this variable really required? Shouldn't the dict comprehensions be sufficient? Also, might this not be better put into the for loop below where we iterate over the data dict anyhow?

Reply:
I initially tried to implement this as a single block of code but I changed to this way as the dataframe sizes could change between before and after the for loop in certain edge cases (I believe if all columns of the dataframe were indices). That is also why I rely on this variable being set instead of `self.data`